server/index.js

```diff
@@ -102,24 +138,49 @@ app.put('/roulette/add', (req, res, next) => {
       SET "details" = $2
       RETURNING *
   `;
-  const params1 = [restaurantId, req.body];
+  const params1 = [req.body.id, req.body];
   db.query(sql1, params1)
     .then(result => {
       const sql2 = `
         INSERT INTO "roulette" ("restaurantId", "accountId")
           VALUES ($1, $2)
           ON CONFLICT ("restaurantId", "accountId") DO NOTHING
       `;
-      const params2 = [restaurantId, TEMP_USER_ID];
+      const params2 = [req.body.id, TEMP_USER_ID];
       db.query(sql2, params2)
         .then(res.status(201).json(result.rows[0]))
         .catch(err => next(err));
     })
     .catch(err => next(err));
 });
 
-app.put('/favorites/add', (req, res, next) => {
-  const { id: restaurantId } = req.body;
+app.delete('/roulette/:id', (req, res, next) => {
+  if (!req.params.id) throw new ClientError(400, 'id is a required field.');
+  const sql1 = `
+    DELETE FROM "roulette"
+      WHERE "restaurantId" = $1
+      AND "accountId" = $2
+      RETURNING *
+  `;
+  const params1 = [req.params.id, TEMP_USER_ID];
+  db.query(sql1, params1)
+    .then(result => {
+      const sql2 = `
+        SELECT "restaurantId"
+          FROM "roulette"
+          WHERE "accountId" = $1
+      `;
+      const params2 = [TEMP_USER_ID];
+      return db.query(sql2, params2);
+    })
```

Comment:
Rather than re-transfer the whole roulette list, I recommend filtering your array on the client when the delete completes